### ISSUE TITLE
Dangling reference to a temporary in OSMFileStore

### ISSUE THREAD

**Issue by SGSSGene** [OpenMS/OpenMS#7060]
gcc13 complains about the following:
```
/home/gene/OpenMS/src/openms/source/FORMAT/OMSFileStore.cpp: In member function ‘void OpenMS::Internal::OMSFileStore::storeConsensusFeatures_(const OpenMS::ConsensusMap&)’:                                                     
/home/gene/OpenMS/src/openms/source/FORMAT/OMSFileStore.cpp:1587:40: warning: possibly dangling reference to a temporary [-Wdangling-reference]                                                                                  
 1587 |         const ConsensusFeature::Ratio& ratio = feat.getRatios()[i];                                                                                                                                                                    
      |                                        ^~~~~                                                                                                                                                                                           
/home/gene/OpenMS/src/openms/source/FORMAT/OMSFileStore.cpp:1587:66: note: the temporary was destroyed at the end of the full expression ‘OpenMS::ConsensusFeature::getRatios() const().std::vector<OpenMS::ConsensusFeature::Rat
io>::operator[](((std::vector<OpenMS::ConsensusFeature::Ratio>::size_type)i))’                                                                                                                                                                 
 1587 |         const ConsensusFeature::Ratio& ratio = feat.getRatios()[i];  
 ```
 From what I can see, this is a correct warning.
 There are two options. We either change `getRatios() const` to return a `const&` to a vector, or we make a copy of ratio:
 `const ConsensusFeature::Ratio ratio = feat.getRatios()[i];`

@hendrikweisser: you have any opinions on this?

**Comment by jpfeuffer** [comment]
It's a bug. I fixed it in one of my upcoming PRs. getRatios needs to return a const ref instead of copies for const objects.
Clang-tidy found it for me.

**Comment by hendrikweisser** [comment]
Thanks for the report and for the fix!

**Comment by SGSSGene** [comment]
This was fixed in commit ce2b1339749f9c0e128ce8424ddd6c752687c7fd .
